### ISSUE TITLE
Using links for Subjects, Author and advisors

### ISSUE THREAD

**Issue by danydvd** [ualbertalib/can-link#48]
@jchartrand the newest SOLR index (CanLink-new-6) now includes subject_url fields. Since this field is multi valued (values are in a list) the subject_url is also a list and each item in this list corresponds to the item in  the subject list. 

`"subject":["roman",
          "inceste",
          "famille",
          "loi",
          "nature",
          "xviiie siècle"],`
`"subject_url":["http://canlink.library.ualberta.ca/subject/b179a9ec0777eae19382c14319872e1b",
"http://canlink.library.ualberta.ca/subject/00e0868463b1ddc2ba31ec9ef835bc35",
"http://canlink.library.ualberta.ca/subject/ed81c0311eb5364a8d6d1d1b63f3c51d",
"http://canlink.library.ualberta.ca/subject/84ab36b2995bb3949db34038a2b24c64",
"http://canlink.library.ualberta.ca/subject/405aaff66082ffe7231d7c1f79926c17",
"http://canlink.library.ualberta.ca/subject/be5ade92ef8c916908660a58e4d36ce0"]`

@sfarnel @jchartrand @CarlsoFiorention what are your thoughts on having a separate page for individual subjects?(e.g. http://206.167.181.124:8000/getitem/http://canlink.library.ualberta.ca/subject/e52aafb06de949864871fd8885da9b04)

**Comment by jchartrand** [comment]
@sfarnel @danydvd 

I'm not totally sure what to do here.

I could hyperlink the subjects that appear on the full record page, e.g.,

http://206.167.181.124/record/e6ef3b77306f69003ece2a41c077840b

so that if you click on a subject it takes you to the corresponding page url from subject_url.

Is that what I should do?

**Comment by sfarnel** [comment]
Thanks @jchartrand I think what we have in mind here (@danydvd please chime in as well!) is that since we have indexed the URIs for the subjects, if we could use that data for matching (in search and facet) but of course display to the end user the text strings then we might get even better results? But perhaps we are already doing this or it maybe harder to do than we think?

**Comment by danydvd** [comment]
I totally agree @sfarnel. If it is feasible that we use indexed URIs (subject_url in SOLR) for search/facet and string value (subject field in SOLR) for display (e.g. http://206.167.181.124:8000/getitem/http://canlink.library.ualberta.ca/thesis/8f2ad2192ac56a61655ecb583a4cd03f)

**Comment by jchartrand** [comment]
@sfarnel @danydvd

I'm for whatever reason really having a hard time with this one.  I guess I don't see how the url versions of the subjects will change matches?

Here is a sample record:

```json
      {
        "id":"http://canlink.library.ualberta.ca/thesis/fb3942f8d9a6f063504ce5a587b6c8a7",
        "year":[2003],
        "title":["Le développement de la personnalité de l'homme de l'adolescence au milieu de la vie : approches centrées sur les variables et sur les personnes"],
        "institution":["http://canlink.library.ualberta.ca/institution/Université_de_Montréal"],
        "institution_name":["Université de Montréal"],
        "institution_short":["U de Montréal"],
        "creator_url":["http://canlink.library.ualberta.ca/person/8ac3c953a43e460d2951dc639cdca8ca"],
        "creator":["Morizot Julien"],
        "lang":["http://id.loc.gov/vocabulary/languages/fre"],
        "link":["http://search.proquest.com/docview/305239136?accountid=12543"],
        "creator_first":["Julien"],
        "creator_last":["Morizot"],
        "degree":["PhD"],
        "degree_name":["Doctor of Philosophy"],
        "subject":["comportement antisocial",
          "judiciarisé",
          "maturation",
          "typologie",
          "trait",
          "continuité"],
        "subject_url":["http://canlink.library.ualberta.ca/subject/433f03787dc72896739f9b371c86cd09",
          "http://canlink.library.ualberta.ca/subject/ee3ce66373ed3b716f25889ffa085445",
          "http://canlink.library.ualberta.ca/subject/b986b2bf13bf3f2657939047c64cec29",
          "http://canlink.library.ualberta.ca/subject/56aa7f225d81ca1bfc651bbf358c4c8e",
          "http://canlink.library.ualberta.ca/subject/94a223decbd8cae91c432ca80c2be1f7",
          "http://canlink.library.ualberta.ca/subject/3442ee7de4438cffaf8c5ea9b3a05f7e"],
        "_version_":1679131393736572928}
```

For that record, how would matching on:

```http://canlink.library.ualberta.ca/subject/433f03787dc72896739f9b371c86cd09```

instead of on the corresponding term:

```comportement antisocial```

change search results?  I feel like I'm just not understanding some piece of all this.

**Comment by sfarnel** [comment]
Thanks @jchartrand I think we may need to do more work on the data end to enable the kind of functionality we had in mind.

**Comment by sfarnel** [comment]
More data work needed; keeping for blue sky purposes
